rest/diagnostic_doc_api.go

```diff
@@ -39,3 +54,53 @@ func (h *handler) handleGetDocChannels() error {
 	h.writeJSON(resp)
 	return nil
 }
+
+// HTTP handler for running a document through the sync function and returning the results
+func (h *handler) handleSyncFnDryRun() error {
+	docid := h.getQuery("docid")
+	revID := h.getQuery("rev")
+
+	body, err := h.readDocument()
```

Comment:
Is it required to specify a body for this call?  If yes, should we return error if it's empty?  If no, does line 64 succeed if not present?

Reply:
if the body is empty and no doc id is not provided it will error, if the body is not provided but a doc id is provided, the doc in the bucket will be used as the body ( will return an error if the doc isnt there).
Ive added test cases for both scenarios